### PR TITLE
Introduced SSL verify config parameter

### PR DESCRIPTION
**Pull request by thierrymarianne**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Allow use of `addons.api_client.verify_ssl` parameter to prevent SSL verification |
| Type? | improvement |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | N/A |
| How to test? | Declare `addons.api_client.verify_ssl` in `parameters.php` to prevent SSL verification |

### CODE REVIEW

File: src/PrestaShopBundle/DependencyInjection/AddOnsConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/DependencyInjection/AddOnsConfiguration.php]: do we need such overkill verification here ? :p

File: tests/Integration/PrestashopBundle/Service/DataProvider/MarketPlace/ApiClientTest.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+namespace PrestaShopBundle\Tests\Service\DataProvider\MarketPlace;
+
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Phake;
+
+/**
+ * @group addons
+ */
+class ApiClientTest extends KernelTestCase
```
Comment (mickaelandrieu) [inline tests/Integration/PrestashopBundle/Service/DataProvider/MarketPlace/ApiClientTest.php]: :+1:  thanks for tests !

**Review by mickaelandrieu — APPROVED** [review]
Cool stuff, maybe too complex... still, I approve :)

File: src/PrestaShopBundle/DependencyInjection/AddOnsConfiguration.php (hunk 1 above)
Comment (thierrymarianne) [inline src/PrestaShopBundle/DependencyInjection/AddOnsConfiguration.php]: Why overkill? Would you rely solely on a parameter? I believe it is pretty standard to validate configuration before relying on it and it provides with sensitive default value.

Comment (mickaelandrieu) [inline src/PrestaShopBundle/DependencyInjection/AddOnsConfiguration.php]: You may check for it only in extension (that you [already do](https://github.com/PrestaShop/PrestaShop/pull/6442/files#diff-95a1b476935b61587950fe999d326bbaR136)).

**Comment by kpodemski** [comment]
It's funny that one additional param means tons of code, i "love" symfony :stuck_out_tongue_winking_eye:

File: src/Core/Addon/Module/ModuleManagerBuilder.php [hunk 3]
```diff
@@ -159,13 +166,18 @@ private function getSymfonyRouter()
...
         $routeFileName = _PS_MODE_DEV_ === true ? 'routing_dev.yml' : 'routing.yml';
-        $routesDirectory = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'app'.DIRECTORY_SEPARATOR.'config';
+        $routesDirectory = $this->getConfigDir();
         $locator = new FileLocator(array($routesDirectory));
         $loader = new YamlFileLoader($locator);
 
         return new Router($loader, $routeFileName);
     }
 
+    protected function getConfigDir()
```
Comment (kpodemski) [inline src/Core/Addon/Module/ModuleManagerBuilder.php]: why method for gettting config dir is in ModuleManager? Shouldn't this be somewhere higher? In some more generic class?

Comment (thierrymarianne) [inline src/Core/Addon/Module/ModuleManagerBuilder.php]: It's a single reuse of strings concatenation. There is no need to have a more generic class (PrestaShop\PrestaShop\Core\Addon\Module\ModuleManager doesn't currently extend any other class).

**Comment by mickaelandrieu** [comment]
LGTM, thank you Thierry :)

**Comment by dSkrbic** [comment]
Hello,

Humm, doesn't work, sam
z1-screenshot_dsk_2016 09 21_15h54m23s_064_
e error

**Comment by kpodemski** [comment]
@dSkrbic did you change `verify` to `false` in `app/config/config.yml`?

**Comment by mickaelandrieu** [comment]
ping @thierrymarianne what do you think ?

**Comment by thierrymarianne** [comment]
@dSkrbic You can also add parameter `addons.api_client.verify_ssl` to `app/config/parameters.php` with  `true` as value

**Pull request merged**